lib/dialects/mssql/data-types.js

```diff
@@ -1,179 +1,197 @@
+    if (!this._binary) {
+      return 'NVARCHAR(' + this._length + ')';
+    } else{
+      return 'BINARY(' + this._length + ')';
     }
-    this.warn('MSSQL does not support TEXT with the `length` option. `NVARCHAR(MAX)` will be used instead.');
-  }
-  return 'NVARCHAR(MAX)';
-};
+  };
+
+  STRING.prototype.escape = false;
+  STRING.prototype.$stringify = function (value, options) {
+    if (this._binary) {
+      return BLOB.prototype.$stringify(value);
+    } else {
+      return options.escape(value);
+    }
+  };
+
+  var TEXT = BaseTypes.TEXT.inherits();
+
+  TEXT.prototype.toSql = function() {
+    // TEXT is deprecated in mssql and it would normally be saved as a non-unicode string.
+    // Using unicode is just future proof
+    if (this._length) {
+      if (this._length.toLowerCase() === 'tiny') { // tiny = 2^8
+        warn('MSSQL does not support TEXT with the `length` = `tiny` option. `NVARCHAR(256)` will be used instead.');
+        return 'NVARCHAR(256)';
+      }
+      warn('MSSQL does not support TEXT with the `length` option. `NVARCHAR(MAX)` will be used instead.');
+    }
+    return 'NVARCHAR(MAX)';
+  };
 
-var BOOLEAN = function() {
-  if (!(this instanceof BOOLEAN)) return new BOOLEAN();
-  BaseTypes.BOOLEAN.apply(this, arguments);
-};
-util.inherits(BOOLEAN, BaseTypes.BOOLEAN);
+  var BOOLEAN = BaseTypes.BOOLEAN.inherits();
 
-BOOLEAN.prototype.toSql = function() {
-  return 'BIT';
-};
+  BOOLEAN.prototype.toSql = function() {
+    return 'BIT';
+  };
 
-BaseTypes.BLOB.prototype.toSql = function() {
-  if (this._length) {
-    if (this._length.toLowerCase() === 'tiny') { // tiny = 2^8
-      this.warn('MSSQL does not support BLOB with the `length` = `tiny` option. `VARBINARY(256)` will be used instead.');
-      return 'VARBINARY(256)';
-    }
-    this.warn('MSSQL does not support BLOB with the `length` option. `VARBINARY(MAX)` will be used instead.');
-  }
-  return 'VARBINARY(MAX)';
-};
+  var UUID = BaseTypes.UUID.inherits();
 
-var UUID = function() {
-  if (!(this instanceof UUID)) return new UUID();
-  BaseTypes.UUID.apply(this, arguments);
-};
-util.inherits(UUID, BaseTypes.UUID);
+  UUID.prototype.toSql = function() {
+    return 'CHAR(36)';
+  };
 
-UUID.prototype.toSql = function() {
-  return 'CHAR(36)';
-};
+  var NOW = BaseTypes.NOW.inherits();
 
-var NOW = function() {
-  if (!(this instanceof NOW)) return new NOW();
-  BaseTypes.NOW.apply(this, arguments);
-};
-util.inherits(NOW, BaseTypes.NOW);
-
-NOW.prototype.toSql = function() {
-  return 'GETDATE()';
-};
+  NOW.prototype.toSql = function() {
+    return 'GETDATE()';
+  };
 
-var DATE = function() {
-  if (!(this instanceof DATE)) return new DATE();
-  BaseTypes.DATE.apply(this, arguments);
-};
-util.inherits(DATE, BaseTypes.DATE);
+  var DATE = BaseTypes.DATE.inherits();
 
-DATE.prototype.toSql = function() {
-  return 'DATETIME2';
-};
+  DATE.prototype.toSql = function() {
+    return 'DATETIME2';
+  };
 
-var INTEGER = function() {
```

Comment:
Nothing changed here - only indentation change by wrapping the whole file in a function